src/nvenc/nvenc_base.cpp

```diff
@@ -457,33 +458,31 @@ namespace nvenc {
   nvenc_base::invalidate_ref_frames(uint64_t first_frame, uint64_t last_frame) {
     if (!encoder || !encoder_params.rfi) return false;
 
-    if (last_frame < first_frame ||
-        encoder_state.last_encoded_frame_index < first_frame ||
-        encoder_state.last_encoded_frame_index > first_frame + 100) {
-      BOOST_LOG(error) << "NvEnc: rfi request " << first_frame << "-" << last_frame << " invalid range (last encoded frame " << encoder_state.last_encoded_frame_index << ")";
-      return false;
-    }
-
-    if (first_frame >= encoder_state.last_rfi_range.first &&
+    if (first_frame == encoder_state.last_rfi_range.first &&
         last_frame <= encoder_state.last_rfi_range.second) {
       BOOST_LOG(debug) << "NvEnc: rfi request " << first_frame << "-" << last_frame << " already done";
       return true;
     }
 
     BOOST_LOG(debug) << "NvEnc: rfi request " << first_frame << "-" << last_frame << " expanding to last encoded frame " << encoder_state.last_encoded_frame_index;
+    last_frame = encoder_state.last_encoded_frame_index;
+
+    if (last_frame < first_frame || last_frame - first_frame + 1 >= encoder_params.ref_frames_in_dpb) {

```

Comment:
`last_frame < first_frame` is legitimately invalid, since it means Moonlight is invalidating a frame it hasn't possibly seen yet. You could log that as an error if you want.

Reply:
Yeah, moved to separate branch with error print.